packages/devtools_app/lib/src/screens/memory/panes/chart/chart_pane.dart

```diff
@@ -0,0 +1,706 @@
+      }
+    });
+
+    addAutoDisposeListener(widget.chartControllers.event.tapLocation, () {
+      if (widget.chartControllers.event.tapLocation.value != null) {
+        if (_hoverOverlayEntry != null) {
+          _hideHover();
+        }
+        final tapLocation = widget.chartControllers.event.tapLocation.value;
+        if (tapLocation?.tapDownDetails != null) {
+          final tapData = tapLocation!;
+          final index = tapData.index;
+          final timestamp = tapData.timestamp!;
+
+          final copied = TapLocation.copy(tapLocation);
+          widget.chartControllers.vm.tapLocation.value = copied;
+          widget.chartControllers.android.tapLocation.value = copied;
+
+          final allValues = ChartsValues(controller, index, timestamp);
+          _showHover(
+            context,
+            allValues,
+            tapData.tapDownDetails!.globalPosition,
+          );
+        }
+      }
+    });
+
+    addAutoDisposeListener(widget.chartControllers.vm.tapLocation, () {
+      if (widget.chartControllers.vm.tapLocation.value != null) {
+        if (_hoverOverlayEntry != null) {
+          _hideHover();
+        }
+        final tapLocation = widget.chartControllers.vm.tapLocation.value;
+        if (tapLocation?.tapDownDetails != null) {
+          final tapData = tapLocation!;
+          final index = tapData.index;
+          final timestamp = tapData.timestamp!;
+
+          final copied = TapLocation.copy(tapLocation);
+          widget.chartControllers.event.tapLocation.value = copied;
+          widget.chartControllers.android.tapLocation.value = copied;
+
+          final allValues = ChartsValues(controller, index, timestamp);
+
+          _showHover(
+            context,
+            allValues,
+            tapData.tapDownDetails!.globalPosition,
+          );
+        }
+      }
+    });
+
+    addAutoDisposeListener(widget.chartControllers.android.tapLocation, () {
+      if (widget.chartControllers.android.tapLocation.value != null) {
+        if (_hoverOverlayEntry != null) {
+          _hideHover();
+        }
+        final tapLocation = widget.chartControllers.android.tapLocation.value;
+        if (tapLocation?.tapDownDetails != null) {
+          final tapData = tapLocation!;
+          final index = tapData.index;
+          final timestamp = tapData.timestamp!;
+
+          final copied = TapLocation.copy(tapLocation);
+          widget.chartControllers.event.tapLocation.value = copied;
+          widget.chartControllers.vm.tapLocation.value = copied;
+
+          final allValues = ChartsValues(controller, index, timestamp);
+
+          _showHover(
+            context,
+            allValues,
+            tapData.tapDownDetails!.globalPosition,
+          );
+        }
+      }
+    });
+
+    addAutoDisposeListener(controller.refreshCharts, () {
+      setState(() {
+        _refreshCharts();
+      });
+    });
+
+    addAutoDisposeListener(controller.androidChartVisibleNotifier, () {
```

Comment:
if we are just calling setState, remove the second parameter here. if the listener is null, it will default to calling setState.

`addAutoDisposeListener(controller.androidChartVisibleNotifier)`